### DISCUSSION TITLE
Migrating from Z-wave JS inside Home Assistant to Node Red

### DISCUSSION THREAD

**Discussion by widert** [zwave-js/node-red-contrib-zwave-js#183] [Q&A] [answered]
Hello, how can I migrate my Z-Wave from Home Assistant to Node Red?

I'm using Z-Wave JS to MQTT and Z-Wave Js.

Is it only to move the USB plug and my network will be recreated?

**Comment by marcus-j-davies** [comment]
Hi @widert,

Please do bear in mind, this module is not designed to work with the internals of HA - its strictly Node RED only.
Most users of this module do not use HA.

On that bases.
make sure to meet the requirements here.
https://github.com/zwave-js/node-red-contrib-zwave-js/wiki/getting-started#before-you-begin

Then after you install the module in Node RED, add a controller node into your flow - select your serial port so on and so forth.
it will start interviewing your network as per normal - so yes, your network will be recreated in Node RED.

or load in the example that comes with the module.

**Reply by widert** [reply]
Perfect, I'll give it a try then!

Tom

**Comment by widert** [comment]
Unfortunately, it seems like it is not able to recreate the network...
![Capture](https://user-images.githubusercontent.com/29115503/154720903-b0b5862d-471b-49c9-b115-9dbea4a3151a.PNG)

**Reply by widert** [reply]
Du I also have to add the encryption keys?

**Reply by marcus-j-davies** [reply]
Hey


it needs to interview all devices.

> Du I also have to add the encryption keys?

if they were added to your controller securely. - yes!.
otherwise the interviews won't succeed

**Reply by marcus-j-davies** [reply]
you can paste your security key in the controller node

**Reply by marcus-j-davies** [reply]
if your key is like 0x00,0x02,0x03 - just paste it  - it will convert to the correct format

**Reply by widert** [reply]
Seems like that helped a bit, and at least one node was created... But still missing a lot....?
![Capture](https://user-images.githubusercontent.com/29115503/154725586-81b3cb2b-0209-4836-910a-d28c286664fc.PNG)

**Reply by marcus-j-davies** [reply]
It's interviewing all your devices.
Node RED doesn't know anything about your devices (Its building it all up).

**Reply by widert** [reply]
Ok, but it seems like it stopped, showing "Not ready"...? 

How long time should this take?

![Capture](https://user-images.githubusercontent.com/29115503/154727265-2bf992a2-a6f3-4005-8624-96c7dfdcb889.PNG)

**Reply by marcus-j-davies** [reply]
OK do this.

1) Make sure you network key(s) are in tact
2) Stop Node Red
3) Delete this folder **~/node-red/zwave-js-cache**
4) Start Node Red

this will be like a clean state for zwave-js.

interviewing can take from 5 minutes to 30 minutes, it really depends on the newtork size and health
when node red starts - don't stop it, or redeploy the controller - it needs to finish.

stopping it may interrupt the process

**Reply by marcus-j-davies** [reply]
Your battery devices won't get interviewed until they wake up.

**Reply by marcus-j-davies** [reply]
each device can take up to 1-3 minutes to interview - again it depends on their signal strength to the controller.

**Reply by widert** [reply]
Well, finally thay appeared. Took some hours... :)

**Reply by marcus-j-davies** [reply]
Cool.

Yeah.... there really isn't a 'it will take this long' answer.
any software needs to know EVERYTHING about the devices, so it knows what devices  can/can't do.

this of course depends heavily on the network.

However, Hopefully you will find this module really nice to use. - most users are surprised at how cool it is.
we are working on v6.6.0 of the module which improves the UI.

adding new devices from hereon in, it is only a few minutes - it's the initial 'rescan' that takes a while.

any questions  - please don't hesitate to ask.

**Reply by widert** [reply]
Thanks!

So, I've managed to create a flow capturing a door sensor... 

And the debug message looks like this:
`msg.payload : Object
object
node: 25
nodeName: "Dørsensor gjesterom"
nodeLocation: "Gjesterom"
event: "VALUE_UPDATED"
timestamp: 1645224635504
object: object
commandClassName: "Notification"
commandClass: 113
endpoint: 0
property: "Access Control"
propertyKey: "Door state"
newValue: 23
prevValue: 22
propertyName: "Access Control"
propertyKeyName: "Door state"`

So I guess it is the "newValue" that I'm looking for to check if the door was opened or closed...?

**Reply by marcus-j-davies** [reply]
Correct.

On the UI - double click the title and you can see what the values relates to.

![image](https://user-images.githubusercontent.com/55892693/154772601-a7bb245c-65e5-439f-89b0-b897d2910566.png)

**Reply by marcus-j-davies** [reply]
Do read up on the **Event-Filter** and **CMD Factory** nodes -  They are very powerful.

https://github.com/zwave-js/node-red-contrib-zwave-js/wiki/node-types


This module also has examples included. - which allows you to see how it all works 👍 

Also - here is the events that you can expect.
https://github.com/zwave-js/node-red-contrib-zwave-js/wiki/Payload-Messaging-Format#event-table

**Reply by widert** [reply]
Perfect, thank you, and great work!

How can I buy you a beer... 😉?

**Reply by marcus-j-davies** [reply]
No need @widert  - appreciation is enough 👍

**Reply by marcus-j-davies** [reply]
You will likely want to update values soon.

EVERY value, property etc etc  has a `ValueID`.
A ValueID is not an ID as such - it's an object, that states the CC, Endpoint and property a value relates to.

Example:

A battery has at least 2 properties, it will therefore have 2 ValueID's.

1 - For the Battery Charge Level
1 - For the Battery Low Status.

You can get a ValueID, buy double clicking a value title, like i shown you above.
the ValueID is everything between `valueId:{......} `.

Example:
to set a binary switch - this assumes you are not using the CMD-Factory

```javascript
let ValueID = {
    "commandClassName": "Binary Switch",
    "commandClass": 37,
    "endpoint": 0,
    "property": "targetValue",
    "propertyName": "targetValue"
}
let Message = {
    "payload": {
        "mode": "ValueAPI",
        "node": 2,
        "method": "setValue",
        "params": [ValueID, false]
    }
}
return Message
```

Battery devices will of course get updated when they awake.
see here, for the ValueID API.

https://github.com/zwave-js/node-red-contrib-zwave-js/wiki/Value-API

hopefully, this will give you a good start.

**Reply by widert** [reply]
Ok, so testing how I can switch outdoor lights on. The switch:
{
   "valueId": {
      "commandClassName": "Binary Switch",
      "commandClass": 37,
      "endpoint": 1,
      "property": "currentValue",
      "propertyName": "currentValue"
   },
   "meta": {
      "type": "boolean",
      "readable": true,
      "writeable": false,
      "label": "Current value"
   },
   "propertyid": "1-37-currentValue-",
   "endpoint": 1,
   "valueData": {
      "value": false
   }
}

So, to switch in on I run the following in a function node:
let ValueID = {
    "commandClassName": "Binary Switch",
    "commandClass": 37,
    "endpoint": 0,
    "property": "targetValue",
    "propertyName": "targetValue"
}
let Message = {
    "payload": {
        "mode": "ValueAPI",
        "node": 14,
        "method": "setValue",
        "params": [ValueID, true]
    }
}
return Message

But unfortunately nothings happens...?

**Reply by marcus-j-davies** [reply]
HI @widert 

you're close.
I think you address the wrong endpoint

![image](https://user-images.githubusercontent.com/55892693/154812407-fa3501c6-df65-401d-a9fd-6d7e097cfca4.png)

This will do what you need it to.
```javascript
let ValueID = {
    "commandClassName": "Binary Switch",
    "commandClass": 37,
    "endpoint": 1, // <----- this is was set to 0, it must match, exactly what the value ID has.
    "property": "targetValue",
    "propertyName": "targetValue"
}

let Message = {
    "payload": {
        "mode": "ValueAPI",
        "node": 14,
        "method": "setValue",
        "params": [ValueID, true]
    }
}
```

Endpoints are used to for example: when a zwave power socket has 2 outlets let's say.

**Reply by widert** [reply]
Hmm, still not working. This is the msg.payload:

msg.payload : Object
object
mode: "ValueAPI"
node: 14
method: "setValue"
params: array[2]
0: object
1: true

**Reply by widert** [reply]
And this is probably the one I should address:

{
   "valueId": {
      "commandClassName": "Binary Switch",
      "commandClass": 37,
      "endpoint": 1,
      "property": "targetValue",
      "propertyName": "targetValue"
   },
   "meta": {
      "type": "boolean",
      "readable": true,
      "writeable": true,
      "label": "Target value",
      "valueChangeOptions": [
         "transitionDuration"
      ]
   },
   "propertyid": "1-37-targetValue-",
   "endpoint": 1,
   "valueData": {
      "value": ""
   }
}

**Reply by marcus-j-davies** [reply]
no worries.

1) Open up UI Monitor

![image](https://user-images.githubusercontent.com/55892693/154812995-04803b73-7eed-4c6f-b88b-43f498e79df5.png)

2) Change the value

![image](https://user-images.githubusercontent.com/55892693/154813027-e85afcfa-fadf-40af-a7d2-c1a903342d6a.png)

3) it will show you the command in UI monitor (or at least it's content), that you can then use 👍 
make sure you are sending this command to the controller node or device node.

If you go to the import menu - there is a full example on how all the nodes work.

**Reply by widert** [reply]
Brilliant, this gave me the following:

{
   "payload": {
      "mode": "ValueAPI",
      "method": "setValue",
      "node": 14,
      "params": [
         {
            "commandClassName": "Binary Switch",
            "commandClass": 37,
            "endpoint": 1,
            "property": "targetValue",
            "propertyName": "targetValue"
         },
         true
      ]
   }
}

**Reply by marcus-j-davies** [reply]
Nice!

The UI will actually control your network so the commands it uses can be used in your function nodes.

**Reply by marcus-j-davies** [reply]
`params` should be `[Your-Value-ID (Object), Your-Value-To-Set]`

**Comment by widert** [comment]
Ok, so next issue is running a roller shutter in/out:

(Testing with CMD Factory)

So to run it down using the method you described earlier, I go into the node, select the Multilevel switch and set Up to true (!)

![image](https://user-images.githubusercontent.com/29115503/154823040-a52c0635-25a2-4a87-a4a7-2fa80e8b4c1f.png)

The UI monitor gives me the following:
![image](https://user-images.githubusercontent.com/29115503/154823080-1b7b3853-f1a6-419e-a63e-089fc3de2914.png)

To replicate I use the following nodes:
![image](https://user-images.githubusercontent.com/29115503/154823260-ab61e8f5-be0f-42de-9ef6-f6363e698c4e.png)


The factory node configuration:
![image](https://user-images.githubusercontent.com/29115503/154823154-304fe04e-61b2-429a-a0f5-b0bd177c2b0c.png)

The Binary switch device:
![image](https://user-images.githubusercontent.com/29115503/154823170-8c2728da-088b-4a9c-a678-d1e3ff95b5a2.png)

The event-filter node:
![image](https://user-images.githubusercontent.com/29115503/154823192-67da03a2-03c9-4e26-9b43-e72fca584814.png)
![image](https://user-images.githubusercontent.com/29115503/154823199-5a5e6415-4633-462e-9f81-46bc0d17af1e.png)

However, when injecting "True", there is a small movement in the roller shutter, but then it stops...??

**Comment by marcus-j-davies** [comment]
Hi @widert,

I think I see the problem.
You have CMD Factory set to use the `CCAPI`, and you are executing the `set` method for `Multilevel Switch`

![image](https://user-images.githubusercontent.com/55892693/154833792-8c03ebc8-79b5-4400-8aa7-8e643d106f0b.png)

Given `Multilevel Switch` is a numerical CC, its `set` method wants a number, its likely taking your Boolean value and converting that to 1 potentially 🤷  

you can see here, the parameters required for `set`
https://zwave-js.github.io/node-zwave-js/#/api/CCs/MultilevelSwitch.

**So, how do we fix this?**
_This is on the assumption you want to use the `CCAPI` and not `ValueAPI`._

**1)** Change the method to `startLevelChange`

![image](https://user-images.githubusercontent.com/55892693/154834856-16e6f264-9d68-4fed-905a-d07d09cb98ac.png)

**2)** change the params text box to the following object (Click the 3 dots ... next to it), and set the endpoint to `topic` for good measure - or what ever property name you want to use.

why? if you look at the required params for `startLevelChange` - it wants an object, so we are constructing said object.
given its `JSONata` we can provide some missing parts from our inject 

![image](https://user-images.githubusercontent.com/55892693/154834993-80757bb1-362f-407a-b0c2-ccd3578a9bcb.png)

```javascript
[
   {
       "ignoreStartLevel": true,
       "direction": payload
   }
]
```

**3)** finally, you can now send using an inject node the below (be sure to set `topic` to a numerical value).
```javascript
{
     payload: "Up" | "Down",     /* Direction */
     topic: 1                    /* Endpoint */
}
```

You can see from the above, the `CCAPI` is extremely verbose - I don't encourage its use.
but the above will do what the UI does but in `CCAPI` mode.

Find below the setup.
```
[{"id":"46d64604fb55147a","type":"cmd-factory","z":"2d7bf6e3.84c97a","name":"ZWave CMD Factory","node":"","endpoint":"topic","cc":"Multilevel Switch","method":"startLevelChange","params":"[{\"ignoreStartLevel\":true,\"direction\":payload}]","noEvent":false,"forceUpdate":"forceUpdate","api":"CCAPI","vapiMode":"setValue","vapiValue":"payload","vapiValueId":"valueId","vapiOptions":"options","x":1040,"y":260,"wires":[["1159ee1f98d465eb"]]},{"id":"afb9961fdf933a29","type":"inject","z":"2d7bf6e3.84c97a","name":"","props":[{"p":"payload"},{"p":"topic","v":"1","vt":"num"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"Up","payloadType":"str","x":830,"y":240,"wires":[["46d64604fb55147a"]]},{"id":"d518ecdd7cf745b4","type":"inject","z":"2d7bf6e3.84c97a","name":"","props":[{"p":"payload"},{"p":"topic","v":"1","vt":"num"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"Down","payloadType":"str","x":830,"y":280,"wires":[["46d64604fb55147a"]]},{"id":"1159ee1f98d465eb","type":"debug","z":"2d7bf6e3.84c97a","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":1270,"y":300,"wires":[]},{"id":"887786e473274856","type":"cmd-factory","z":"2d7bf6e3.84c97a","name":"ZWave CMD Factory","node":"","endpoint":"topic","cc":"Multilevel Switch","method":"stopLevelChange","params":"","noEvent":false,"forceUpdate":"","api":"CCAPI","vapiMode":"setValue","vapiValue":"payload","vapiValueId":"valueId","vapiOptions":"options","x":1040,"y":340,"wires":[["1159ee1f98d465eb"]]},{"id":"9d6650c46cb8b139","type":"inject","z":"2d7bf6e3.84c97a","name":"Stop","props":[{"p":"payload"},{"p":"topic","v":"1","vt":"num"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":830,"y":340,"wires":[["887786e473274856"]]}]
```

If you are addressing this to a device node, depending on its mode, you can omit `node` in your message.

My suggestion however - set CMD Factory to `ValueAPI` mode its just easier 😉 

`CCAPI` does not sync the local data cache - Unless the device updates us.
you have to do `get` commands if you invest in the `CCAPI` - the CCAPI is very "manual"

is CCAPI useful? -  yes if course, but more often than not, its only used if you are executing things that the `ValueAPI` does not provide.

Hopefully the above make sense? 😬

**Comment by marcus-j-davies** [comment]
Using CMD Factory is trivial when using `ValueID` Mode.
payload will just be `true` | `false` (or what ever value is needed)

you can pass in `ValueID` from a message part if you wish `msg.valueID`, but just ensure it matches what the UI used.
As the `valueID` has the `endpoint` - if you specify it here also, it will just overwrite the one in the `ValueID`

![image](https://user-images.githubusercontent.com/55892693/154837636-647787cf-1334-4d8b-9b27-b60ae85ad737.png)

![image](https://user-images.githubusercontent.com/55892693/154837839-09954d36-97d0-481e-b925-7fdb4dc8a575.png)

You can also provide a partial `ValueID` and substitute the `property` in the ValueID to set it to `Up` or `Down` based on  `msg.direction` (Again, its JSONata)

```javascript
{
   "commandClassName": "Mulitlevel Switch",
   "commandClass": 38,
   "endpoint": 1,
   "property": direction,
   "propertyName": direction
}
```

So many possibilities.
this module is a framework - it allows being bent into shape that best works for you. 👍

**Reply by widert** [reply]
Puuh, struggelig to get this right. 

I tested the CCAPI example, but was not able to make it work.

Using your suggestion this is how my Command factory node is configured:
![image](https://user-images.githubusercontent.com/29115503/154866375-61ab8ab6-b4e3-4526-829c-f69d69021be7.png)

Debug:
![image](https://user-images.githubusercontent.com/29115503/154866487-2af2f72f-ad87-4d99-a0a9-ec26e604a1b3.png)

**Comment by widert** [comment]
So if I instead would like to go the ValueAPI route then I need to send the following:

![image](https://user-images.githubusercontent.com/29115503/154866695-edcf3693-77c3-4550-b19f-f78e803aa75b.png)

How do I send this? Through a function node only?

**Reply by marcus-j-davies** [reply]
Hi @widert 

You're  connecting the CMD Factory to the device-node or controller right? 😄 

The `CCAPI` method should work, but some devices want a starting level, and you will provide that in your payload (under the same object that `direction` and `ignoreStartLevel` is under)

the property is called `startLevel`

`[{"ignoreStartLevel": false, "direction":"Up", "startLevel": <Number>}]`

The ValueAPI is likely adding this property behind the scene, hence why it works under the UI (The UI uses the ValueIAPI)

see here, for the ValueAPI Method using CMD Factory.
```
[{"id":"46d64604fb55147a","type":"cmd-factory","z":"2d7bf6e3.84c97a","name":"ZWave CMD Factory","node":"","endpoint":"","cc":"Multilevel Switch","method":"startLevelChange","params":"[{\"ignoreStartLevel\":true,\"direction\":payload}]","noEvent":false,"forceUpdate":"forceUpdate","api":"ValueAPI","vapiMode":"setValue","vapiValue":"payload","vapiValueId":"{\t   \"commandClassName\": \"Mulitlevel Switch\",\t   \"commandClass\": 38,\t   \"endpoint\": 1,\t   \"property\": direction,\t   \"propertyName\": direction\t}","vapiOptions":"","x":1040,"y":260,"wires":[["073971ea6f261507"]]},{"id":"afb9961fdf933a29","type":"inject","z":"2d7bf6e3.84c97a","name":"Up","props":[{"p":"payload"},{"p":"direction","v":"Up","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"true","payloadType":"bool","x":830,"y":240,"wires":[["46d64604fb55147a"]]},{"id":"073971ea6f261507","type":"debug","z":"2d7bf6e3.84c97a","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":1290,"y":260,"wires":[]},{"id":"112b8202bc929ed2","type":"inject","z":"2d7bf6e3.84c97a","name":"Down","props":[{"p":"payload"},{"p":"direction","v":"Down","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"true","payloadType":"bool","x":810,"y":320,"wires":[["46d64604fb55147a"]]}]
```

if you wanted to do it in a function node;
```javascript
let ValueID = {
    "commandClassName": "Mulitlevel Switch",
    "commandClass": 38,
    "endpoint": 1,
    "property": "Up",
    "propertyName": "Up"
}

let Message = {
    payload: {
        "mode": "ValueAPI",
        "method": "setValue",
        "node": 10,
        "params": [ValueID, true]
    }
}

return Message
```

**Reply by widert** [reply]
Ok, finally got it running now using the following flow;
![image](https://user-images.githubusercontent.com/29115503/154867655-e2282257-50c8-473f-bcd2-3777c421b9e6.png)

My function node contains:
![image](https://user-images.githubusercontent.com/29115503/154867568-c5068fdf-c890-41bb-9d54-832d81db6a9c.png)

**Reply by marcus-j-davies** [reply]
Cool!
It sounds like your device needs a starting level (some don't)
the starting level is added behind the scene when using ValueAPI - but you can do it via CCAPI also as per my previous comment.

In essence, the CMD Factory node in ValueAPI mode will do exactly what the function node is doing.

ValueAPI is the better API

👍

**Comment by marcus-j-davies** [comment]
You can now see, how your function node, has the shape of the message displayed in UI Monitor.

![](https://user-images.githubusercontent.com/29115503/154866695-edcf3693-77c3-4550-b19f-f78e803aa75b.png)

We are just wrapping the valueID in a variable, instead of writing it in line.

```javascript
let ValueID = {
    "commandClassName": "Mulitlevel Switch",
    "commandClass": 38,
    "endpoint": 1,
    "property": "Up",
    "propertyName": "Up"
}

let Message = {
    payload: {
        "mode": "ValueAPI",
        "method": "setValue",
        "node": 10,
        "params": [ValueID, true]
    }
}

return Message
```

**Reply by widert** [reply]
How can I now read status of the blinds, i.e if they are up or down?

The "Current Value" doesn't change, maybe I need to run calibration first?

![image](https://user-images.githubusercontent.com/29115503/154904370-9fc9f474-d6f4-4590-a958-1f39b34f9737.png)

**Comment by marcus-j-davies** [comment]
Morning @widert 

You can obtain the current Value with the below.
All we are doing is swopping `setValue` for `getValue`
and we want the ValueID for `currentValue`

```javascript
let ValueID = {
    "commandClassName": "Mulitlevel Switch",
    "commandClass": 38,
    "endpoint": 1,
    "property": "currentValue",
    "propertyName": "currentValue"
}
let Message = {
    "payload": {
        "mode": "ValueAPI",
        "node": 10,
        "method": "getValue",
        "params": [ValueID]
    }
}
return Message
```

but also, when ever the device sends an update you will get a `VALUE_UPDATED` event. from the node

**Reply by widert** [reply]
So, a lot of questions, I'm sorry...
But how can I read the state (valueData?) out of the followoing:

```
{
   "valueId": {
      "commandClassName": "Notification",
      "commandClass": 113,
      "endpoint": 0,
      "property": "Home Security",
      "propertyKey": "Motion sensor status",
      "propertyName": "Home Security",
      "propertyKeyName": "Motion sensor status"
   },
   "meta": {
      "type": "number",
      "readable": true,
      "writeable": false,
      "min": 0,
      "max": 255,
      "label": "Motion sensor status",
      "states": {
         "0": "idle",
         "8": "Motion detection"
      },
      "ccSpecific": {
         "notificationType": 7
      }
   },
   "propertyid": "0-113-Home Security-Motion sensor status",
   "endpoint": 0,
   "valueData": {
      "value": 0
   }
}
```

**Comment by marcus-j-davies** [comment]
I had a notification, but cant find anything you wrote?
did you solve it?

**Reply by widert** [reply]
Tried again... :)

**Reply by marcus-j-davies** [reply]
No worries.

it's the same for obtaining the roller status, but for a different ValueID. 👍 

```javascript
let ValueID = {
    "commandClassName": "Notification",
      "commandClass": 113,
      "endpoint": 0,
      "property": "Home Security",
      "propertyKey": "Motion sensor status",
      "propertyName": "Home Security",
      "propertyKeyName": "Motion sensor status"
}
let Message = {
    "payload": {
        "mode": "ValueAPI",
        "node": <NodeID>,
        "method": "getValue",
        "params": [ValueID]
    }
}
return Message
```

**Reply by marcus-j-davies** [reply]
And don't be Sorry.

**Reply by marcus-j-davies** [reply]
And remember, you will also get them, when they update. as a `VALUE_UPDATED` event.

**Comment by widert** [comment, marked as answer]
Hmm, not getting it right. My setup:
![image](https://user-images.githubusercontent.com/29115503/155034222-c9e596da-9f6b-43a9-aaf3-b089b8009af8.png)

The function
```
let ValueID = {
    "commandClassName": "Notification",
      "commandClass": 113,
      "endpoint": 0,
      "property": "Home Security",
      "propertyKey": "Motion sensor status",
      "propertyName": "Home Security",
      "propertyKeyName": "Motion sensor status"
}
let Message = {
    "payload": {
        "mode": "ValueAPI",
        "node": 11,
        "method": "getValue",
        "params": [ValueID]
    }
}
return Message
```

The z-wave device mode:

![image](https://user-images.githubusercontent.com/29115503/155034368-ea06a4f3-1495-492e-baea-280be249dd3f.png)

The event-filter node:

![image](https://user-images.githubusercontent.com/29115503/155034483-9853fa81-539e-4209-9582-9c6a3fab3f72.png)

The filter:
```
[
    {
        "commandClassName": "Notification",
        "commandClass": 113,
        "endpoint": 0,
        "property": "Home Security",
        "propertyKey": "Motion sensor status",
        "propertyName": "Home Security",
        "propertyKeyName": "Motion sensor status"
    }
]
```

The response I'm getting:
```
msg.payload : Object
object
node: 11
nodeName: "Multisensor gang"
nodeLocation: "Gang"
event: "GET_VALUE_RESPONSE"
timestamp: 1645483385059
object: object
response: 0
valueId: object
commandClassName: "Notification"
commandClass: 113
endpoint: 0
property: "Home Security"
propertyKey: "Motion sensor status"
propertyName: "Home Security"
propertyKeyName: "Motion sensor status"
```

**Reply by marcus-j-davies** [reply]
Its here.

`response: 0` 😄

**Reply by marcus-j-davies** [reply]
You don't have to use event filter - its just a helper tool, if you want to split the values out to different paths/wires

**Reply by marcus-j-davies** [reply]
Like this.

![image](https://user-images.githubusercontent.com/55892693/155035216-777fb2e8-ceed-48e7-9dc0-de286dae9026.png)

**Reply by widert** [reply]
Thanks!

Too late in the evening...😁
